mesonbuild/dependencies/pkgconfig.py

```diff
@@ -32,64 +32,144 @@
-                   '{!r}'.format(name, self.pkgbin.get_path()))
-        ret, self.version, _ = self._call_pkgbin(['--modversion', name])
+    def found(self) -> bool:
+        return bool(self.pkgbin)
+
+    def version(self, name: str) -> T.Optional[str]:
+        mlog.debug(f'Determining dependency {name!r} with pkg-config executable {self.pkgbin.get_path()!r}')
+        ret, version, _ = self._call_pkgbin(['--modversion', name])
+        return version if ret == 0 else None
+
+    @staticmethod
+    def _define_variable_args(define_variable: T.Optional['ImmutableListProtocol[str]']) -> T.List[str]:
+        if define_variable:
+            return ['--define-variable=' + '='.join(define_variable)]
+        return []
+
+    def cflags(self, name: str, allow_system: bool = False,
+               define_variable: T.Optional['ImmutableListProtocol[str]'] = None) -> T.List[str]:
+        env = None
+        if allow_system:
+            env = os.environ.copy()
+            env['PKG_CONFIG_ALLOW_SYSTEM_CFLAGS'] = '1'
+        args: T.List[str] = []
+        args += self._define_variable_args(define_variable)
+        args += ['--cflags', name]
+        ret, out, err = self._call_pkgbin(args, env=env)
         if ret != 0:
-            return
+            raise DependencyException(f'Could not generate cflags for {name}:\n{err}\n')
+        return self._split_args(out)
 
-        self.is_found = True
+    def libs(self, name: str, static: bool = False, allow_system: bool = False,
+             define_variable: T.Optional['ImmutableListProtocol[str]'] = None) -> T.List[str]:
+        env = None
+        if allow_system:
+            env = os.environ.copy()
+            env['PKG_CONFIG_ALLOW_SYSTEM_LIBS'] = '1'
+        args: T.List[str] = []
+        args += self._define_variable_args(define_variable)
+        if static:
+            args.append('--static')
+        args += ['--libs', name]
+        ret, out, err = self._call_pkgbin(args, env=env)
+        if ret != 0:
+            raise DependencyException(f'Could not generate libs for {name}:\n{err}\n')
+        return self._split_args(out)
+
+    def variable(self, name: str, variable_name: str,
+                 define_variable: 'ImmutableListProtocol[str]') -> T.Optional[str]:
+        args: T.List[str] = []
+        args += self._define_variable_args(define_variable)
+        args += ['--variable=' + variable_name, name]
+        ret, out, err = self._call_pkgbin(args)
+        if ret != 0:
+            raise DependencyException(f'could not get variable for {name}:\n{err}\n')
+        variable = out.strip()
+        # pkg-config doesn't distinguish between empty and nonexistent variables
+        # use the variable list to check for variable existence
+        if not variable:
+            ret, out, _ = self._call_pkgbin(['--print-variables', name])
+            if not re.search(r'^' + variable_name + r'$', out, re.MULTILINE):
```

Comment:
```suggestion
            if not re.search(f'^{variable_name}$', out, re.MULTILINE):
```

Reply:
Might as well prefix it with `r` too

Reply:
> Might as well prefix it with `r` too, assuming that is possible

Why? There are no special characters to escape here....

Reply:
I just always prefix regex strings with r regardless

Reply:
done